A tutorial on how to create a 3Box chatbox plugin.md

```diff
@@ -10,3 +10,119 @@
  5.How to build with 3box chatbox plugins
  6.Reasons for using 3box chatbox plugins
 
+
+A TUTORIAL ON HOW TO CREATE A 3BOX CHATBOX PLUGIN
+
+
+# INTRODUCTION TO 3BOX CHATBOX PLUGIN
+
+## What is Chatbox
+Chatbox is a 3Box plugin which enables dapp developers to add chatrooms to their react Ethereum application with a few lines of code to enable rich social discourse. The 3Box Chatbox plugin is built using 3Box infrastructure, 3Box Ghost Threads and handles all 3Box and web3 logic for creating a chat box. 3Box is creating a rich plugin ecosystem that simplifies the web3 development experience.
+
+## Ghost threads
+Ghost threads are a different type of messaging thread, that do not persist messages to OrbitDB database storage. Instead, ghost messages are sent from one peer to other peers on the network using IPFS pubsub, and are kept in-memory upon receipt by online peers. New users coming online or joining the chatroom can request a message backlog from other users in the chat. The Ghost Threads API allows Ethereum app developers to create new chat features that demand scalable performance to support many users sending many messages such as site-wide chat rooms, lightning fast load speeds, or novel properties such as disappearing messages.
+
+# USE CASES OF 3BOX CHATBOX PLUGIN
+
+Asset-driven use cases are a great start, and native to blockchains, most activity on mainstream applications today is more social and less financial in nature. Chatbox provide simple social functionality that drives everyday engagement on dapps Below are some available use cases.
+
+## Decide on a pop-up or in-page component.
+Chatbox can configured to be displayed as a pop-up modal or as an embedded in-page element depending on the user experience you wish to provide.
+
+## Swap in your app’s name, image/logo, and color.
+Chatbox, with its simple drop-in functionality, makes it easier than ever to experiment with social web3 development. it’s fully-featured out-of-the-box and can scale with your application into production.
+
+# HOW 3BOX CHATBOT WORKS
+
+## Architecture
+The Chatbox plugin is built using a standard implementation of 3Box Ghost Threads which are defined in the 3Box Threads API and made available via the 3Box.js SDK. As with all Ghost Threads, Chatbox messages are sent from one peer to other peers presently connected to the network using IPFS/libp2p pubsub and are then stored in-memory by online peers. The message backlog is persisted as long as there is at least one user in the chatbox, however if all users go offline the history will disappear. 
+The Chatbox plugin includes UI for an embedded in-window or pop-up chat room along with all relevant logic. The component is configurable to various authentication patterns, and can handle both Web3/3Box logged-in and logged-out states.
+
+## Authentication
```

Comment:
@writeprovidence Same as above, this is mostly just a copy and paste, either add your own thoughts, or just link to this section